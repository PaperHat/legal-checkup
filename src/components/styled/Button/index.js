import styled from 'styled-components'

const Button = styled.button`
  background-color: #eee;
  color: #333;
  width: 88px;
  height: 78px;
`
Button.displayName = 'StyledButton'

export default Button
