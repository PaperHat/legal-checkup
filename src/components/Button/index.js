import React, { Fragment } from 'react'

import { StyledButton } from '../styled'

export default function Button () {
  return (
    <Fragment>
      <StyledButton>{this.props.color}</StyledButton>
    </Fragment>
  )
}
