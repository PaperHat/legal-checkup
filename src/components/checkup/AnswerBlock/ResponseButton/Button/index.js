import styled from 'styled-components'

import {
  BACKGROUND_BUTTON,
  BUTTON_BORDER_FOCUS,
  BLACK,
  DESKTOP,
  MOBILE,
  TABLET,
  WHITE,
  ROUGE,
  BACKGROUND_BUTTON_ACTIVE,
  OFF_WHITE,
  TEXT_COLOR
} from '../../../../../constants'
import styledMap from 'styled-map'

import { NOT_SURE } from '../../../../../state/constants'

function setWidthButton (format, type) {
  if (format === DESKTOP || format === TABLET) {
    return type === NOT_SURE ? '155px' : '153px'
  } else {
    return type === NOT_SURE ? '105px' : '85px'
  }
}

function setMarginRight (format, type) {
  if (format === DESKTOP || format === TABLET) {
    return type === NOT_SURE ? 0 : '10px'
  } else {
    return type === NOT_SURE ? 0 : '7px'
  }
}

function setPreviousSelectedBackgroundColor (buttonType, activeQuestionAnswer) {
  return buttonType === activeQuestionAnswer
    ? BACKGROUND_BUTTON_ACTIVE
    : OFF_WHITE
}

function setPreviousSelectedTextColor (buttonType, activeQuestionAnswer) {
  return buttonType === activeQuestionAnswer ? ROUGE : TEXT_COLOR
}

const Button = styled.button`
  background-color: ${({ format, type, activeQuestionAnswer }) =>
    setPreviousSelectedBackgroundColor(type, activeQuestionAnswer)};
  color: ${({ format, type, activeQuestionAnswer }) =>
    setPreviousSelectedTextColor(type, activeQuestionAnswer)};
  height: 78px;
  border: none;
  font-family: 'Museo Sans Cyrl 500', sans-serif;
  font-weight:500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  width: ${({ format, type }) => setWidthButton(format, type)};
  margin-right: ${({ format, type }) => setMarginRight(format, type)};,

  &:after {
    content: '${({ type }) => type}';
  }

  &:hover {
    background-color: ${BACKGROUND_BUTTON};
    color: ${WHITE};
  }
  &:active {
    background-color: ${BLACK};
    color: ${WHITE};
  }
  &:focus{
    outline: 0px;
    border-color: ${BUTTON_BORDER_FOCUS}
    border-style: solid;
  }
  &:focus:active {
    background-color: ${BLACK};
    color: ${WHITE};
  }

  ${styledMap('format', {
    [DESKTOP]: `
    border-radius: 5px;
    font-size:26px;`,
    [MOBILE]: `
    border-radius: 4px;
    font-size:20px;
    margin-bottom: 20px`,
    [TABLET]: `
    border-radius: 5px;
    font-size:26px;`
  })}
`
Button.displayName = 'Button'

export default Button
