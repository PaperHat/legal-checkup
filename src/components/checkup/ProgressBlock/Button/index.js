import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

const getColor = ({ active, enabled }) => {
  if (active) {
    return 'white'
  } else if (enabled) {
    return 'black'
  } else {
    return '#c0c0c0'
  }
}

const Button = styled.button`
  stroke: ${({ enabled }) => (enabled ? '#b62645' : '#dcb1b8')};
  border: none;
  color: ${getColor};
  cursor: ${({ enabled }) => (enabled ? 'pointer' : 'default')};
  border-radius: 50%;
  background-color: ${({ active }) => (active ? '#b62645' : 'transparent')};
  font-size: 13px;
  font-weight: bold;
  outline: none;
  width: 20px;
  height: 20px;
  line-height: 20px;
  text-align: center;
  padding: 0px;

  ${styledMap('format', {
    [`${DESKTOP}Next`]: ``,
    [`${DESKTOP}Previous`]: ``,
    [`${DESKTOP}Question`]: ``,
    [`${MOBILE}Next`]: ``,
    [`${MOBILE}Previous`]: ``,
    [`${MOBILE}Question`]: ``,
    [`${TABLET}Next`]: ``,
    [`${TABLET}Previous`]: ``,
    [`${TABLET}Question`]: ``
  })}
`
Button.displayName = 'Button'

export default Button
