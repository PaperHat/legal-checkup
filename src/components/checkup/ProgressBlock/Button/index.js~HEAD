import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

const Button = styled.button`
  // background-color: #ededed;
  border-radius: 0 5px 5px 0;
  border: none;
  color: ${({ enabled }) => (enabled ? '#000' : '#c0c0c0')};
  // color: ${({ enabled, active }) => (enabled && active ? 'white' : '#c0c0c0')};
  height: 30px;
  cursor: ${enabled => (enabled ? 'pointer' : 'default')}
  border-radius: ${({ active }) => (active ? '22px' : '0 5px 5px 0')};
  background-color: ${({ active }) => (active ? '#b62645' : '#ededed')};
  font-weight: ${({ enabled }) => (enabled ? 'bold' : 'normal')};
  margin-top: 30px;

  outline: none;
  width: 30px;

  ${styledMap('format', {
    [`${DESKTOP}Next`]: ``,
    [`${DESKTOP}Previous`]: ``,
    [`${DESKTOP}Question`]: ``,
    [`${MOBILE}Next`]: ``,
    [`${MOBILE}Previous`]: ``,
    [`${MOBILE}Question`]: ``,
    [`${TABLET}Next`]: ``,
    [`${TABLET}Previous`]: ``,
    [`${TABLET}Question`]: ``
  })}

`
Button.displayName = 'Button'

export default Button
