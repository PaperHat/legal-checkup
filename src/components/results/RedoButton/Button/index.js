import styled from 'styled-components'
import styledMap from 'styled-map'

import {
  DESKTOP,
  MOBILE,
  TABLET,
  BACKGROUND_BUTTON,
  WHITE
} from '../../../../constants'

const Button = styled.div`
  border-radius: 5px;
  border: solid 3px #ffffff;
  font-family: 'Museo Sans Cyrl 900', sans-serif;
  font-size: 20px;
  text-align: center;
  color: #b62645;
  text-decoration: none;
  margin: auto;

  &:hover {
    background-color: ${BACKGROUND_BUTTON};
    color: ${WHITE};
  }

  ${styledMap('format', {
    [DESKTOP]: `
      height: 78px;
      padding-top: 39px;
      width: 920px;
    `,
    [MOBILE]: `
      height: 24px;
      padding: 28px 26px 26px;
    `,
    [TABLET]: `
      height: 24px;
      padding: 28px 26px 26px;
    `
  })}
`
Button.displayName = 'Button'

export default Button
