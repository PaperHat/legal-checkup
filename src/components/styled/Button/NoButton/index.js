import Button from '..'

const NoButton = Button.extend`
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
`

NoButton.displayName = 'NoButton'

export default NoButton
