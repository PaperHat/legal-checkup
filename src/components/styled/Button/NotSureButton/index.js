import Button from '..'

const NotSureButton = Button.extend`
  margin-left: 7px;
  width: 109px;
  height: 78px;
  border-radius: 5px;
  border: solid 2px #b5b5b5;
`

NotSureButton.displayName = 'NoButton'

export default NotSureButton
