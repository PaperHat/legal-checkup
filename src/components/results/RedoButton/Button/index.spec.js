import * as React from 'react'
import { create } from 'react-test-renderer'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

import Button from './'

describe('components:results:RedoButton:Button', function () {
  it(`matches the snapshot when format is ${DESKTOP}`, () => {
    expect(create(<Button format={DESKTOP} />).toJSON()).toMatchSnapshot()
  })

  it(`matches the snapshot when format is ${MOBILE}`, () => {
    expect(create(<Button format={MOBILE} />).toJSON()).toMatchSnapshot()
  })

  it(`matches the snapshot when format is ${TABLET}`, () => {
    expect(create(<Button format={TABLET} />).toJSON()).toMatchSnapshot()
  })
})
