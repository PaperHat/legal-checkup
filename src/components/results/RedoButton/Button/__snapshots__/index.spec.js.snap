// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:results:RedoButton:Button matches the snapshot when format is desktop 1`] = `
.c0 {
  border-radius: 5px;
  border: solid 3px #ffffff;
  font-family: Helvetica;
  font-size: 20px;
  font-weight: 900;
  text-align: center;
  color: #b62645;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: auto;
  height: 78px;
  padding-top: 39px;
  width: 920px;
}

<div
  className="c0"
  format="desktop"
/>
`;

exports[`components:results:RedoButton:Button matches the snapshot when format is mobile 1`] = `
.c0 {
  border-radius: 5px;
  border: solid 3px #ffffff;
  font-family: Helvetica;
  font-size: 20px;
  font-weight: 900;
  text-align: center;
  color: #b62645;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: auto;
  width: 310px;
  height: 24px;
  padding: 28px 26px 26px;
}

<div
  className="c0"
  format="mobile"
/>
`;

exports[`components:results:RedoButton:Button matches the snapshot when format is tablet 1`] = `
.c0 {
  border-radius: 5px;
  border: solid 3px #ffffff;
  font-family: Helvetica;
  font-size: 20px;
  font-weight: 900;
  text-align: center;
  color: #b62645;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: auto;
  width: 310px;
  height: 24px;
  padding: 28px 26px 26px;
}

<div
  className="c0"
  format="tablet"
/>
`;
